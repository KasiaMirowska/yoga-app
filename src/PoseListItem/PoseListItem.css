.item {
    background-color: white; 
    margin: 25px;
    text-align: center;
    padding: 20px;
    border-radius: 10px;
    filter: drop-shadow(0 0 0.75rem rgb(151, 132, 136));
}


@media (min-width: 600px) {
    .item {
        width: 33%; 
    }
    
  }  