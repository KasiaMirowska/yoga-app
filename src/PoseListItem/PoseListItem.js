import React from 'react';
import TokenService from '../services/token-service';
import config from '../config';
import { withRouter } from 'react-router-dom';



function PoseListItem(props) {
  
    const handleClick = () => {
        const token = TokenService.hasAuthToken(config.TOKEN_KEY)
            if (!token) {
                console.log('NO TOKEN')
                props.history.push(`/login`)
            } 
            else if(props.location.pathname =='/'){
                props.history.push('/flowForm')
            }
            else {
                props.history.push(`/flow/${props.id}`)
            }
    } 
   
    return (
        <div onClick={()=>handleClick()}>
                <h2>{props.name}</h2>
                <h2>{props.saskrit}</h2>
                <img src={props.img} />
           
       </div>
       
    )
}

export default withRouter(PoseListItem);