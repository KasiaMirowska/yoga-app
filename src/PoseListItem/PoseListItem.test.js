import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import PoseListItem from './PoseListItem';
import {BrowserRouter as BR} from 'react-router-dom';

describe('PoseListItem component', () => {
    const props = {
        name: 'chair',
        sanskrit: 'utkatasana',
        img: '',
        location: {pathname: '/'},
        history: {push : '/'}
    };

    it('renders without crashing', () => {
         const div = document.createElement('div');
         ReactDom.render(<BR><PoseListItem {...props} /></BR>, div);
         ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const item = renderer.create(<BR><PoseListItem {...props}/></BR>)
        expect(item.toJSON()).toMatchSnapshot();
    });
});
