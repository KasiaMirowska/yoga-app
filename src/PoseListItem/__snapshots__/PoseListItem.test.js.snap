// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoseListItem component renders UI as expected 1`] = `
<div
  onClick={[Function]}
>
  <h2>
    chair
  </h2>
  <h2 />
  <img
    src=""
  />
</div>
`;
